### PR TITLE
swev-id: django__django-16662 - group MigrationWriter imports by type

### LINKED ISSUE #518: Fix migration import ordering to match isort and Django style
User's request:

Migration import ordering violates coding style and isort defaults

Description
New migration files are generated with imports sorted by module, independent of import style. For example:

import datetime
from django.db import migrations, models
import time

The Django coding style specifies:
Place all import module statements before from module import objects in each section.
This guidance is the same as what isort does by default, as documented here. Newly generated migrations can fail isort for this reason.
This would mean migration files should instead be generated like this:

import datetime
import time
from django.db import migrations, models

For reference, previous issues related to migration import sorting: #24155, #25384.

---

Researcher specification (Emerson Gray):
- Code path: django/django/db/migrations/writer.py, class MigrationWriter, method as_string(). Current sort mixes 'import' and 'from' by module name only.
- Change: Group imports as future imports first (if any), then plain 'import X', then 'from X import Y'; within each group sort by module (alpha). Keep existing logic merging 'from django.db import models' into 'from django.db import migrations, models'.
- Proposed snippet:
  - future_imports = sorted([i for i in imports if i.startswith('from __future__ import ')], key=lambda i: i.split()[1])
  - plain_imports = sorted([i for i in imports if i.startswith('import ')], key=lambda i: i.split()[1])
  - from_imports = sorted([i for i in imports if i.startswith('from ') and not i.startswith('from __future__ import ')], key=lambda i: i.split()[1])
  - sorted_imports = future_imports + plain_imports + from_imports
  - items['imports'] = '\n'.join(sorted_imports) + '\n' if imports else ''
- Edge cases: future imports (top), TYPE_CHECKING not used, keep merged imports behavior.
- Tests: Add tests/migrations/test_writer.py::test_imports_grouped_plain_before_from asserting:
  import datetime\nimport time\nfrom django.db import migrations, models\n
- Reproduction: Prior behavior produces:
  import datetime\nfrom django.db import migrations, models\nimport time\n
  isort --check-only --diff flags this and moves 'import time' above 'from django.db ...'.

Acceptance:
- One PR and one Issue for this request; PR targets branch django__django-16662 and includes title token 'swev-id: django__django-16662'. Do not merge.

### PR DESCRIPTION
**Pull request by casey-brooks**
Fixes #518.

## Reproduction and observed failure (on django__django-16662)
1. Check out `django__django-16662`.
2. Run:
   ```sh
   PYTHONPATH=$PWD:$HOME/.nix-profile/lib/python3.11/site-packages DJANGO_SETTINGS_MODULE=tests.test_sqlite python3 - <<'PY'
   import datetime
   import time

   from django.db import migrations, models
   from django.db.migrations.writer import MigrationWriter

   migration = type(
       "Migration",
       (migrations.Migration,),
       {
           "operations": [
               migrations.CreateModel(
                   "TimedModel",
                   [
                       ("id", models.AutoField(primary_key=True)),
                       ("created_at", models.DateTimeField(default=datetime.datetime.now)),
                       ("duration", models.FloatField(default=time.time)),
                   ],
                   {},
                   (models.Model,),
               )
           ]
       },
   )
   writer = MigrationWriter(migration)
   print(writer.as_string().split("\n\n", 2)[1])
   PY
   ```
3. The generated import block is:
   ```
   import datetime
   from django.db import migrations, models
   import time
   ```
4. Save the snippet above into `tmp_imports.py` and run:
   ```sh
   PYTHONPATH=$HOME/.local/lib/python3.11/site-packages:$HOME/.nix-profile/lib/python3.11/site-packages python3 -m isort --check-only --diff tmp_imports.py
   ```
   Output before this change:
   ```
   ERROR: /workspace/django/tmp_imports.py Imports are incorrectly sorted and/or formatted.
   --- /workspace/django/tmp_imports.py:before	2025-12-24 17:33:25.490772
   +++ /workspace/django/tmp_imports.py:after	2025-12-24 17:33:28.788401
   @@ -1,4 +1,4 @@
    import datetime
   -from django.db import migrations, models
    import time
   
   +from django.db import migrations, models
   ```

## Summary
- group migration imports into future, plain, and from blocks so plain imports precede from-imports
- retain the combined `from django.db import migrations, models` special-case handling
- add `WriterTests.test_imports_grouped_plain_before_from` covering the new ordering

## Testing
- `PYTHONPATH=$PWD:$HOME/.nix-profile/lib/python3.11/site-packages DJANGO_SETTINGS_MODULE=tests.test_sqlite python3 tests/runtests.py migrations.test_writer --parallel=1 --verbosity=1`
- `PYTHONPATH=$PWD:$HOME/.nix-profile/lib/python3.11/site-packages flake8 django/db/migrations/writer.py tests/migrations/test_writer.py`

### CODE REVIEW

**Comment by casey-brooks** [comment]
## Local Verification
- `PYTHONPATH=$PWD:$HOME/.nix-profile/lib/python3.11/site-packages DJANGO_SETTINGS_MODULE=tests.test_sqlite python3 tests/runtests.py migrations.test_writer --parallel=1 --verbosity=1` → 53 passed, 0 failed, 0 skipped
- `PYTHONPATH=$PWD:$HOME/.nix-profile/lib/python3.11/site-packages flake8 django/db/migrations/writer.py tests/migrations/test_writer.py` → lint clean (no issues)

**Review by noa-lucent — APPROVED** [review]
Changes look good. The new import grouping logic matches the requested ordering and the added test covers the regression scenario.
